core,native: persist native contract via VM

fyrchik opened the pull request:
Closes #1060 .

After native contracts are deployed, single persist script is created
and executed at the start of every block persisting.

roman-khimov commented on pkg/core/blockchain.go:
```diff
@@ -576,6 +576,8 @@ func (bc *Blockchain) storeBlock(block *block.Block) error {
 		if bc.config.FreeGasLimit > 0 {
 			v.GasLimit = bc.config.FreeGasLimit + tx.SystemFee
+		} else {
+			v.GasLimit = tx.SystemFee
```
I think we may not touch it until #1064.

roman-khimov commented on pkg/core/blockchain.go:
```diff
@@ -556,6 +556,15 @@ func (bc *Blockchain) storeBlock(block *block.Block) error {
+		v := SpawnVM(systemInterop)
+		v.LoadScriptWithFlags(bc.contracts.GetPersistScript(), smartcontract.AllowModifyStates|smartcontract.AllowCall)
+		if err := v.Run(); err != nil {
+			bc.log.Warn("can't persist native contracts", zap.Error(err))
```
I think we should fail persisting the block then. And there also should be a wrapper cached DAO for these cases. In general, it's not that different from transaction execution handling.

fyrchik replied:
Fixed.

fyrchik replied:
What do you mean by `wrapper cached DAO`? We use already created cache and `interop.NewContext` takes care about wrapping.

roman-khimov replied:
OK.

roman-khimov replied:
But I don't see it being persisted though.

fyrchik replied:
Oh, yeah, fixed.

fyrchik commented:
Added `AppExecResult` saving.
Dropped a comment about extending `getapplicationlog` in #1062 .
I am not sure it is needed in event stream, as I see no usecase and it will be always the same (besides consumed gas):
`HALT`, 2 `true` items on stack (# of native contracts), `trigger.System`.

roman-khimov commented:
But there will be notifications on burned/minted gas and we need them.

roman-khimov commented:
So what about `bc.processNEP5Transfer`?

fyrchik commented:
@roman-khimov I am working on it. RN `block.ConsensusData.PrimaryIndex` is not set properly in tests which makes it hard to make changes in RPC tests.
